autoload/xtabline/cmds.vim

```diff
@@ -388,6 +388,26 @@ fun! s:reopen_last_tab()
     if !a:0 | call xtabline#update() | endif
   endfun
 
+  fun! s:move_buffer_next()
+    let b = bufnr("%") | let oB = s:oB() | let max = len(oB) - 1
+    let i = index(oB, b)
+    if i + 1 > max
+      call s:move_buffer(0)
+    else
+      call s:move_buffer(i + 1)
+    endif
+  endfun
+
+  fun! s:move_buffer_prev()
+    let b = bufnr("%") | let oB = s:oB() | let max = len(oB) - 1
+    let i = index(oB, b)
+    if i - 1 < 0
+      call s:move_buffer(max)
```

Comment:
If `i < 0` it means the buffer doesn't appear in the tabline, `oB` is a list of buffers that are show there, in the order you see them (=`orderedBuffers`), so if it isn't there, you can't move it. Moreover there's no way to call these functions from anywhere.

If I understood, you want a command to move a buffer to next/previous position, and not to a precise position with [count], right?

Reply:
Thank you for replay :smile:

I want to move the buffer without [count]. It means moving the buffer by continuing key input like tabmove +/-.

https://vim-jp.org/vimdoc-en/tabpage.html#:tabmove

*Usage*

```
nnoremap <C-F2> :<C-u>call xtabline#cmds#run('move_buffer_prev')<CR>
nnoremap <C-F3> :<C-u>call xtabline#cmds#run('move_buffer_next')<CR>
```